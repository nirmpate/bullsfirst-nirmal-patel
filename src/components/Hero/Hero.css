.hero {
  display: grid;
  grid-template-columns: 2fr 1fr;
}

.hero-bg {
  background-color: var(--neutral-50);
}

.hero h1 {
  margin-top: 40px;
}

.hero-img {
  max-width: 190px;
}

.hero-btn {
  align-self: center;
}

.hero-btn button {
  float: right;
}

.popsicle-container img {
  max-width: 66px;
  max-height: 200px;
  transform: rotate(25deg);
  filter: drop-shadow(2px 8px 2px rgba(0, 0, 0, 0.06))
    drop-shadow(0px 4px 4px rgba(0, 0, 0, 0.35));
}

.popsicle-container {
  position: absolute;
  right: 190px;
  bottom: -120px;
  display: none;
}

@media (min-width: 640px) {
  .hero {
    grid-template-columns: 1fr 1fr;
  }

  .popsicle-container {
    display: block;
  }
}
