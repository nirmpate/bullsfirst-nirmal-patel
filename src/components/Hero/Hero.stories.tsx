import { Meta, Story } from '@storybook/react';
import { Hero, HeroType } from './Hero';
export default {
    title: 'Components/HeroComponent',
    component: Hero,
    argTypes: {
        onSignIn: {action: 'clicked'}
    }
} as Meta;

const Template: Story<HeroType> = (args: HeroType) => <Hero {...args} />

export const HeroComponentStory = Template.bind({});

