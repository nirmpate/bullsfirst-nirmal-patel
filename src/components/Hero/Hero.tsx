import logo from '../../assets/bullsfirst-logo.svg';
import './Hero.css';
import { Button } from '../Button';

interface HeroType {
    onSignIn: () => {}
}

export const Hero = (props: HeroType) => {

    const onClick: any = () => {
        return props.onSignIn();
    };

    return (
        <div className="hero-bg">
            <div className="hero container">
                <div>
                    <img className="hero-img" src={ logo } alt="Bullsfirst Logo"/>
                    <h1 className="title">
                        Get better results with Bullsfirst at the helm of your portfolio.
                    </h1>
                </div>
                <div className="hero-btn">
                    <Button click={() => onClick()} variant="filled" color="primary" type="button" linkText="Sign In"/>
                </div>
            </div>
        </div>

    )
  };