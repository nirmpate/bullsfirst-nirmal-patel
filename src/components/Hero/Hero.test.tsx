import { fireEvent, render, screen } from '@testing-library/react';
import { Hero } from './Hero';


test('renders correctly', () => {
  const handleClick = jest.fn;
  const { asFragment } = render(<Hero onSignIn={handleClick} />);
  expect(asFragment()).toMatchSnapshot();
});


test('calls onClick', () => {
    const handleClick = jest.fn();
    render(<Hero onSignIn={handleClick} />);
    const button = screen.getByText('Sign In');
    fireEvent.click(button);
    expect(handleClick).toBeCalled();
  });