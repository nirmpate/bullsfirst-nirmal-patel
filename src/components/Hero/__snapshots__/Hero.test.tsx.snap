// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="hero-bg"
  >
    <div
      class="hero container p-relative"
    >
      <div>
        <img
          alt="Bullsfirst Logo"
          class="hero-img"
          src="bullsfirst-logo.svg"
        />
        <h1
          class="title"
        >
          Get better results with Bullsfirst at the helm of your portfolio.
        </h1>
      </div>
      <div
        class="hero-btn"
      >
        <button
          class="button button--filled button--primary"
        >
           Sign In
        </button>
      </div>
      <div
        class="popsicle-container"
      >
        <img
          alt="Popsicle"
          src="popsicle.png"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
